os/net/ipv6/sicslowpan.c

```diff
@@ -1639,25 +1630,16 @@ output(const linkaddr_t *localdest)
     set_packet_attrs();
   }
 
-  /*
-   * The destination address will be tagged to each outbound
-   * packet. If the argument localdest is NULL, we are sending a
-   * broadcast packet.
-   */
-  if(localdest == NULL) {
-    linkaddr_copy(&dest, &linkaddr_null);
-  } else {
-    linkaddr_copy(&dest, localdest);
-  }
-
   LOG_INFO("output: sending IPv6 packet with len %d\n", uip_len);
 
   /* copy over the retransmission count from uipbuf attributes */
   packetbuf_set_attr(PACKETBUF_ATTR_MAX_MAC_TRANSMISSIONS,
                      uipbuf_get_attr(UIPBUF_ATTR_MAX_MAC_TRANSMISSIONS));
 
-/* Calculate NETSTACK_FRAMER's header length, that will be added in the NETSTACK_MAC */
-  packetbuf_set_addr(PACKETBUF_ADDR_RECEIVER, &dest);
+  /* Copy destination address to packetbuf */
+  packetbuf_set_addr(PACKETBUF_ADDR_RECEIVER,
+      localdest ? localdest : &linkaddr_null);
```

Comment:
Should we be more defensive and rather use `localdest == NULL`? Avoiding any future- or portability issues.

Reply:
I think that is a style issue, the definition of `NULL` is part of the C standard so a compiler breaking that will not work for a lot of code out there.

Reply:
Thx! I did not realize it was in the standard.

Reply:
I could actually use the elvis operator here (`localdest ?: &linkaddr_null`)

Reply:
Had to google that one :) I don't have any strong opinion either way.